docs/design/values.md

```diff
@@ -0,0 +1,1027 @@
+
+**Future work:** Add explicit designs for these use cases and link to them here.
+
+### Reference types
+
+Unlike C++, Carbon does not currently have reference types. The only form of
+indirect access are pointers. There are a few aspects to this decision that need
+to be separated carefully from each other as the motivations and considerations
+are different.
+
+First, Carbon has only a single fundamental construct for indirection because
+this gives it a single point that needs extension and configuration if and when
+we want to add more powerful controls to the indirect type system such as
+lifetime annotations or other safety or optimization mechanisms. The designs
+attempts to identify a single, core indirection tool and then layer other
+related use cases on top. This is motivated by keeping the language scalable as
+it evolves and reducing the huge explosion of complexity that C++ sees due to
+having a large space here. For example, when there are N > 1 ways to express
+indirection equivalently and APIs want to accept any one of them across M
+different parameters they can end up with N \* M combinations.
+
+Second, with pointers, Carbon's indirection mechanism retains the ability to
+refer distinctly to the point*er* and the point*ee* when needed. This ends up
+critical for supporting rebinding and so without this property more permutations
+of indirection would likely emerge.
+
+Third, Carbon doesn't provide a straightforward way to avoid the syntactic
+distinction between indirect access and direct access.
+
+For a full discussion of the tradeoffs of these design decisions, see the
+alternatives considered section of [P2006]:
+
+-   [References in addition to pointers](/proposals/p2006.md#references-in-addition-to-pointers)
+-   [Syntax-free or automatic dereferencing](/proposals/p2006.md#syntax-free-or-automatic-dereferencing)
+-   [Exclusively using references](/proposals/p2006.md#exclusively-using-references)
+
+### Pointer syntax
+
+The type of a pointer to a type `T` is written with a postfix `*` as in `T*`.
+Dereferencing a pointer is a [_reference expression_] and is written with a
+prefix `*` as in `*p`:
+
+```carbon
+var i: i32 = 42;
+var p: i32* = &i;
+
+// Form a reference expression `*p` and assign `13` to the referenced storage.
+*p = 13;
+```
+
+This syntax is chosen specifically to remain as similar as possible to C++
+pointer types as they are commonly written in code and are expected to be
+extremely common and a key anchor of syntactic similarity between the languages.
+The different alternatives and tradeoffs for this syntax issue were discussed
+extensively in [#523] and are summarized in the
+[proposal](/proposals/p2006.md#alternative-pointer-syntaxes).
+
+[#523]: https://github.com/carbon-language/carbon-lang/issues/523
+
+Carbon also supports an infix `->` operation, much like C++. However, Carbon
+directly defines this as an exact rewrite to `*` and `.` so that `p->member`
+becomes `(*p).member` for example.
```

Comment:
It took me a moment to realize what important detail this "However" was describing, in contrast to C++. Maybe it's worth spelling out that in Carbon there is no overloaded `operator->`, and instead customizing the behavior of `*p` is what governs the behavior of `->`?